ecommerce-ui/src/App.js

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import Button from './components/Button';
+import ShoppingCart from './components/ShoppingCart';
+import VacationRental from './components/VacationRental';
+import bnbs from './data/bnbs.json';
+let totalCost = 0;
+function App() {
+  const [itemList, setItemList] = useState([]);
+  const [toggleDisplay, setToggleDisplay] = useState(true);
+  // Add the selected vacation rental to the itemList array
+  const addItem = (id) => {
+    setItemList([...itemList, bnbs[id]]);
+    // Sum total cost
+    totalCost += bnbs[id].payment.cost;
```

Comment:
While this works here, I would make sure that `totalCost` is stored within the component via `useState`. This will guarantee that changing the value will indeed re-render the component and display the right amount. In this case, it works, but it's possible for future changes to break this functionality.